app/talk/discussion.cjsx

```diff
@@ -49,6 +46,18 @@ module.exports = React.createClass
     if @props.params.discussion isnt nextProps.params.discussion
       @setDiscussion(nextProps.params.discussion)
         .then => @setComments(nextProps.location.query.page ? 1)
+        .then =>
+          if nextProps.user isnt @props.user
+            talkClient
```

Comment:
Probably should be catching errors, though. If the roles call fails, the only side effect I can think of is that you don't get the banner under your username beside the comment box.